### PR TITLE
Cherry-pick improved ProposalView and fix AgreementView

### PR DESCRIPTION
**Pull request by nieznanysprawiciel**

### CODE REVIEW

File: agent/provider/src/market/negotiator/builtin/max_agreements.rs [hunk 1]
```diff
@@ -38,7 +38,7 @@ impl NegotiatorComponent for MaxAgreements {
         } else {
             log::info!(
                 "'MaxAgreements' negotiator: Reject proposal [{}] due to limit.",
-                demand.agreement_id, // TODO: Should be just `id`, but I reuse AgreementView struct.
+                demand.id, // TODO: Should be just `id`, but I reuse AgreementView struct.
```
Comment (MrDarthShoe) [inline agent/provider/src/market/negotiator/builtin/max_agreements.rs:41]: Probably not needed
```suggestion
                demand.id,
```

File: utils/agreement-utils/src/agreement.rs [hunk 2]
```diff
@@ -322,6 +209,7 @@ pub fn try_from_path(path: &PathBuf) -> Result<Value, Error> {
         None => DEFAULT_FORMAT,
     };
 
+    eprintln!("Parsing agreement at {}", path.display());
```
Comment (MrDarthShoe) [inline utils/agreement-utils/src/agreement.rs:212]: Probably some leftover: `log::trace`

File: utils/agreement-utils/src/agreement.rs [hunk 3]
```diff
@@ -11,15 +17,19 @@ pub const PROPERTY_TAG: &str = "@tag";
...
+//  as ProposalView, but as long as we don't merge them, we need to keep them.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
 pub struct AgreementView {
     pub json: Value,
-    pub agreement_id: String,
+    pub id: String,
 }
 
+pub type OfferView = ProposalView;
+pub type DemandView = ProposalView;
```
Comment (MrDarthShoe) [inline utils/agreement-utils/src/agreement.rs:31]: What are these for? Probably ujnused

**Review by MrDarthShoe — COMMENTED** [review]
Besides some probably leftovers, looks very good!

File: utils/agreement-utils/src/agreement.rs (hunk 2 above)
Comment (nieznanysprawiciel) [inline utils/agreement-utils/src/agreement.rs:212]:
```suggestion
```

File: utils/agreement-utils/src/agreement.rs (hunk 3 above)
Comment (nieznanysprawiciel) [reply utils/agreement-utils/src/agreement.rs:31]: I think I'm using them in incoming negotiator changes

**Review by MrDarthShoe — APPROVED** [review]

**Review by MrDarthShoe — APPROVED** [review]

**Pull request merged**